app/controllers/settings/profiles_controller.rb

```diff
@@ -14,7 +14,7 @@ class Settings::ProfilesController < ApplicationController
   def show; end
 
   def update
-    if @account.update(account_params)
+    if UpdateAccountService.new.call(@account, account_params)
```

Comment:
You should check if a similiar change is needed in app/controllers/api/v1/accounts/credentials_controller.rb (update via API)